src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -599,16 +596,18 @@ namespace OpenMS
           const AASequence ion = peptide.getPrefix(i);
           addIsotopeCluster_(spectrum, ion, ion_names, charges, res_type, charge, intensity);
         }
-      }
+        chunks.add(true);
 
-      if (add_losses_ && add_isotopes_) // otherwise losses are already added, see above
-      {
-        // add loss peaks (slow)
-        Size i = add_first_prefix_ion_ ? 1 : 2;
-        for (; i < peptide.size(); ++i)
+        if (add_losses_)
         {
-          const AASequence ion = peptide.getPrefix(i);
-          addLosses_(spectrum, ion, ion_names, charges, intensity, res_type, charge);
+          // add loss peaks (slow)
+          Size i = add_first_prefix_ion_ ? 1 : 2;
```

Comment:
As you did it above:
`Size i = Size(!add_first_prefix_ion) + 1`

Reply:
Hm, my solution is kind of hacky. In this case one could also say `i = 2 - add_first_prefix_ion_;` but if think the ternary operator is still the most readable. So, I'm gonna leave my solution for `0 : 1` cases and leave "more complex" ones to the ternary operator.